reconcile/terraform_resources.py

```diff
@@ -428,6 +432,11 @@ def run(
             dry_run, oc_map, ri, thread_pool_size, caller=acc_name
         )
 
+    # update secrets in vault immediately after applying them to the clusters
+    # so that nothing can raise an error before the secrets are updated
+    if actions and vault_output_path:
+        write_outputs_to_vault(vault_output_path, resource_specs)
```

Comment:
i would use `defer` here to avoid so that nothing can raise an error before disabling keys

Reply:
Here we only moved `write_outputs_to_vault` up, how about other steps after `disable_keys`, those cleanup steps can be skipped too.

Reply:
As @hemslo said, it's about making `write_outputs_to_vault` more bulletproof. In the worst, worst, worst case, `disabled_keys` would run at the next `terraform-resources` reconciliation loop but it runs in `aws-iam-keys` anyways. It also doesn't depend on the current run, but `write_outputs_to_vault` does.

@hemslo, there isn't anything after `disabled_keys`

Reply:
the runs of aws-iam-keys is different if it runs from terraform-resources.

Reply:
Ah, I see 👍 , but anyway, a missed `write_outputs_to_vault` is worse than a missed `disabled_keys`.

Reply:
if only we had something that will allow us not to miss anything, like `defer`.

Reply:
if `disabled_keys` throw error, then `cleanup_and_exit` won't run

Reply:
if we expect `disable_keys` to throw error, how about catch it and process it similar to `ri.has_error_registered`?